crates/tools/src/main.rs

```diff
@@ -44,5 +44,21 @@ fn install_code_extension() -> Result<()> {
     } else {
         run(r"code --install-extension ./ra-lsp-0.0.1.vsix --force", "./editors/code")?;
     }
+    verify_installed_extensions()?;
+    Ok(())
+}
+
+fn verify_installed_extensions() -> Result<()> {
+    let exts = if cfg!(windows) {
+        run_with_output(r"cmd.exe /c code.cmd --list-extensions", ".")?
+    } else {
+        run_with_output(r"code --list-extensions", ".")?
+    };
+    if String::from_utf8(exts.stdout)?.lines().find(|x| x.contains("ra-lsp")).is_none() {
```

Comment:
I think this could be replaced with `std::str::from_utf8(&exts.stdout)` to avoid the allocation. In addition I think we can avoid the `lines` and just use `contains` on the string itself.

Reply:
Thanks, `lines` isn't needed at all. `Vec<u8>` is already allocated in `Output` and `String::from_utf8` just copies the `Vec` triplet but `str::from_utf8` might indeed be cheaper.